Some trivial XCB bugs

Found with clang static code analyzer:

https://github.com/LaurentGomila/SFML/blob/master/src/SFML/Window/Unix/InputImpl.cpp#L205
```
src/SFML/Window/Unix/InputImpl.cpp:205:31: warning: Value stored to 'result' is never read
        case Mouse::Left:     result = pointer->mask & XCB_BUTTON_MASK_1;
                              ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SFML/Window/Unix/InputImpl.cpp:206:31: warning: Value stored to 'result' is never read
        case Mouse::Right:    result = pointer->mask & XCB_BUTTON_MASK_3;
                              ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SFML/Window/Unix/InputImpl.cpp:207:31: warning: Value stored to 'result' is never read
        case Mouse::Middle:   result = pointer->mask & XCB_BUTTON_MASK_2;
                              ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

https://github.com/LaurentGomila/SFML/blob/master/src/SFML/Window/Unix/WindowImplX11.cpp#L538
```
src/SFML/Window/Unix/WindowImplX11.cpp:538:73: warning: Use of memory after it is freed
        err() << "Failed to set the window's icon: Error code " << (int)errptr->error_code << std::endl;
                                                                        ^~~~~~~~~~~~~~~~~~
```

I'm not sure if this is the right place to put this, but I ran into an XCB related issue trying to build my project with the latest master branch for SFML. Here's the output I got using CodeBlocks with static release libraries.

`/usr/bin/ld: /dev/SFML-build/lib/libsfml-window-s.a(WindowImplX11.cpp.o)||undefined reference to symbol 'XSetEventQueueOwner'|`

I'm using Ubuntu 14.04. I just installed all the libxcb-xxxx-dev packages that seemed to be required for CMake.

--- a/src/SFML/Window/Unix/InputImpl.cpp
+++ b/src/SFML/Window/Unix/InputImpl.cpp
@@ -194,24 +194,21 @@ bool InputImpl::isMouseButtonPressed(Mouse::Button button)
 
     // Get pointer mask
     xcb_query_pointer_reply_t* pointer = xcb_query_pointer_reply(connection, xcb_query_pointer(connection, XDefaultRootWindow(display)), NULL);
+    uint16_t mask = pointer->mask;
+    free(pointer);
 
     // Close the connection with the X server
     CloseDisplay(display);
 
-    bool result = false;
-
     switch (button)
     {
-        case Mouse::Left:     result = pointer->mask & XCB_BUTTON_MASK_1;
-        case Mouse::Right:    result = pointer->mask & XCB_BUTTON_MASK_3;
-        case Mouse::Middle:   result = pointer->mask & XCB_BUTTON_MASK_2;
-        case Mouse::XButton1: // not supported by X
-        case Mouse::XButton2: // not supported by X
-        default:              result = false;
+        case Mouse::Left:     return mask & XCB_BUTTON_MASK_1;
+        case Mouse::Right:    return mask & XCB_BUTTON_MASK_3;
+        case Mouse::Middle:   return mask & XCB_BUTTON_MASK_2;
+        case Mouse::XButton1: return false; // not supported by X
+        case Mouse::XButton2: return false; // not supported by X
+        default:              return false;
     }
-
-    free(pointer);
-    return result;
 }
 
 

--- a/src/SFML/Window/Unix/WindowImplX11.cpp
+++ b/src/SFML/Window/Unix/WindowImplX11.cpp
@@ -530,12 +530,10 @@ void WindowImplX11::setIcon(unsigned int width, unsigned int height, const Uint8
     xcb_free_gc(m_connection, iconGC);
 
     xcb_generic_error_t* errptr = xcb_request_check(m_connection, cookie);
-    bool setWindowIconFailed = (errptr != NULL);
-    free(errptr);
-
-    if (setWindowIconFailed)
+    if (errptr)
     {
-        err() << "Failed to set the window's icon: Error code " << (int)errptr->error_code << std::endl;
+        err() << "Failed to set the window's icon: Error code " << static_cast<int>(errptr->error_code) << std::endl;
+        free(errptr);
         return;
     }
 

